### PR TITLE
Add CI confing for client-pkg repository

### PR DESCRIPTION
**Pull request by dsimansk**
**What this PR does, why we need it**:

There's a new repository to share code between `kn` client and plugins. This PR enables CI jobs etc.

- Add CI confign for `client-pkg` repository

/cc @rhuss

### CODE REVIEW

File: config/prod/prow/config_knative.yaml [hunk 1]
```diff
@@ -277,6 +277,11 @@ presubmits:
     always-run: true
     command:
     - ./test/presubmit-integration-tests-latest-release.sh
+  knative/client-pkg:
+  - build-tests: true
+  - unit-tests: true
+  - integration-tests: true
```
Comment (dsimansk) [inline config/prod/prow/config_knative.yaml:283]: @rhuss I'd say that we don't necessarily need integration tests for client-pkg. Wdyt? We should be fine with build and unit tests.

Comment (rhuss) [reply config/prod/prow/config_knative.yaml:283]: yes, there is not really something we could test integration wise. Maybe the watch functionality, but I would add integration tests when we would need it the first time. let's start small.

File: config/prod/prow/config_knative.yaml [hunk 2]
```diff
@@ -277,6 +277,11 @@ presubmits:
...
+  - go-coverage: false
```
Comment (rhuss) [inline config/prod/prow/config_knative.yaml:284]: no coverage ?

File: config/prod/prow/config_knative.yaml (hunk 1 above)
Comment (dsimansk) [reply config/prod/prow/config_knative.yaml:283]: Or maybe we'll keep it blank and come up with something meaningful? :)

File: config/prod/prow/config_knative.yaml [hunk 3]
```diff
@@ -896,6 +901,10 @@ periodics:
     - INGRESS_CLASS="contour.ingress.networking.knative.dev"
     external_cluster:
       secret: s390x-cluster1
+  knative/client:
```
Comment (rhuss) [inline config/prod/prow/config_knative.yaml:904]: was this missing before ?

Comment (dsimansk) [reply config/prod/prow/config_knative.yaml:904]: :facepalm: that's a copy paste error.

File: config/prod/prow/config_knative.yaml (hunk 2 above)
Comment (dsimansk) [reply config/prod/prow/config_knative.yaml:284]: We'll have still have Codecov and only skip the go-coverage from Prow.

E.g.: https://github.com/knative/test-infra/blob/main/config/prod/prow/config_knative.yaml#L24

**Review by rhuss — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**